### PR TITLE
chore(core): improve cleanup method's reliability and fix several bad calls

### PR DESCRIPTION
**Pull request by bziobrowski**
This PR makes Misc.free() and Misc.freeObjList() accept Closeable-derived argument only to improve type safety . 
If input type is not necessarily a Closeable then freeIfCloseable() or freeObjListIfCloseable() should be used . 
PR also contains fixes for several bad calls of aforementioned methods .

### CODE REVIEW

**Comment by bluestreak01** [comment]
this PR should be a "chore" as it is not user-facing issue

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 40 / 50 (80.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/MapSymbolColumn.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/LocalValueMap.java|1|7|14.29%|
|:large_blue_circle:|io/questdb/std/Misc.java|8|10|80.00%|
|:large_blue_circle:|io/questdb/cairo/TableReader.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/std/AssociativeCache.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameSequence.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/std/WeakMutableObjectPool.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/PageAddressCacheRecord.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/SymbolColumn.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpServer.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameReduceJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/log/LogFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/analytic/RowNumberFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/mp/WorkerPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/ThreadLocal.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/EmptyTableRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/FilterOnExcludedValuesRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/mp/RingQueue.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/WalWriter.java|2|2|100.00%|

File: core/src/main/java/io/questdb/cairo/TableWriter.java [hunk 1]
```diff
@@ -2751,15 +2751,15 @@ private void doClose(boolean truncate) {
         Misc.free(other);
         Misc.free(todoMem);
         Misc.free(attachMetaMem);
-        Misc.free(attachMetadata);
+        //Misc.free(attachMetadata);
```
[the remark this answers is not in the document]
Comment (bziobrowski) [reply core/src/main/java/io/questdb/cairo/TableWriter.java]: I left them earlier for Miguel to review . Removed them now .

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 40 / 50 (80.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/MapSymbolColumn.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/LocalValueMap.java|1|7|14.29%|
|:large_blue_circle:|io/questdb/std/Misc.java|8|10|80.00%|
|:large_blue_circle:|io/questdb/cairo/TableReader.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/std/AssociativeCache.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameSequence.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/std/WeakMutableObjectPool.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/PageAddressCacheRecord.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/SymbolColumn.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpServer.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameReduceJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/log/LogFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/analytic/RowNumberFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/mp/WorkerPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/ThreadLocal.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/EmptyTableRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/FilterOnExcludedValuesRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/mp/RingQueue.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/WalWriter.java|2|2|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**